Docs: Add environment variable reference page (#2131)
ROCm/hipBLASLt: docs/reference/env-variables.rst

.. meta::
  :description: hipBLASLt environment variables reference
  :keywords: hipBLASLt, ROCm, API, environment variables, environment, reference

.. _environment-variables:

********************************************************************
hipBLASLt environment variables
********************************************************************

This section describes the important hipBLASLt environment variables,
which are grouped by functionality.

> Logging and Debugging

Reviewer: ```suggestion
Logging and debugging
```

=====================

The logging and debugging environment variables for hipBLASLt are collected in the following table.
For more information, see :doc:`Use logging and heuristics <../how-to/use-logging-heuristics>`.

.. list-table::
    :header-rows: 1
    :widths: 70,30

    * - **Environment variable**
      - **Value**

    * - | ``HIPBLASLT_LOG_LEVEL``
        | Controls the verbosity level of hipBLASLt logging output.
      - | 0: Off (logging disabled, default)
        | 1: Error (only errors are logged)
        | 2: Trace (API calls with kernel launches log parameters)
        | 3: Hints (performance improvement suggestions)
        | 4: Info (general library execution information)
        | 5: API trace (detailed API call parameters)

    * - | ``HIPBLASLT_LOG_MASK``
        | Controls logging output using bit mask flags (can be combined).
      - | 0: Off
        | 1: Error
        | 2: Trace
        | 4: Hints
        | 8: Info
        | 16: API trace
        | 32: Bench
        | 64: Profile
        | 128: Extended profile

    * - | ``HIPBLASLT_LOG_FILE``

>         | Specifies path to logging file. Can contain %i for process ID replacement.

Reviewer: ```suggestion
        | Specifies path to logging file. Can contain ``%i`` for process ID replacement.
```

>       - | Path to log file (e.g., "logfile_%i.log")

Reviewer: ```suggestion
      - | Path to log file (for example, ``logfile_%i.log``)
```

        | If not defined: log messages printed to stdout

    * - | ``HIPBLASLT_ENABLE_MARKER``
        | Enables marker trace for ROCProfiler profiling.

>       - | 1: Enable marker trace

Reviewer: Typically low-to-high values elsewhere (here high-to-low)

        | 0 or unset: Disable marker trace

> Offline Tuning

Reviewer: ```suggestion
Offline tuning
```

===============

The offline tuning environment variables for hipBLASLt are collected in the following table.
For more information, see :doc:`Use hipBLASLt offline tuning <../how-to/how-to-use-hipblaslt-offline-tuning>`.

.. list-table::
    :header-rows: 1
    :widths: 70,30

    * - **Environment variable**
      - **Value**

    * - | ``HIPBLASLT_TUNING_FILE``
        | Specifies file to store tuning results with best solution indices for GEMM problems.

>       - | Path to tuning file (e.g., "tuning.txt")

Reviewer: ```suggestion
      - | Path to tuning file (for example, ``tuning.txt``)
```

        | File stores optimal kernel indices for reuse

    * - | ``HIPBLASLT_TUNING_OVERRIDE_FILE``
        | Specifies file to load tuning results and override default kernel selection.

>       - | Path to tuning file (e.g., "tuning.txt")

Reviewer: ```suggestion
      - | Path to tuning file (for example, ``tuning.txt``)
```

        | Loads previously saved optimal kernel choices

    * - | ``HIPBLASLT_TUNING_USER_MAX_WORKSPACE``
        | Sets maximum workspace size constraint during tuning stage.
      - | Integer value in bytes (default: 128 * 1024 * 1024)
        | Limits workspace size for solution selection

> Stream-K Configuration

Reviewer: ```suggestion
Stream-K configuration
```
